Formula/hyeong.rb

```diff
@@ -0,0 +1,15 @@
+class Hyeong < Formula
+  desc "Hyeo-ung Programming Language Compiler in Rust (beta release)"
+  homepage "https://github.com/buttercrab/hyeo-ung-lang"
+  url "https://github.com/buttercrab/hyeo-ung-lang/releases/download/v0.1.0/hyeong-0.1.0-darwin-x86-64.tar.gz"
+  version "0.1.0"
+  sha256 "24990f87afe0813a0d67ba6525bc822b73f470c58692b5eedcdb5203dec2aa63"
+
+  def install
+    bin.install "hyeong"
+  end
+
+  test do
+    assert_match "hyeong 0.1.0", shell_output("#{bin}/hyeong --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.